aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -0,0 +1,477 @@
+            });
+        });
+```
+
+For more information, see the following articles:
+
+* <xref:fundamentals/configuration/index>
+* <xref:fundamentals/host/generic-host>
+
+## Kestrel
+
+* Kestrel configuration has been updated for the migration to the Generic Host. In 3.0, Kestrel is configured on the web host builder provided by `ConfigureWebHostDefaults`.
+* Connection Adapters have been removed from Kestrel and replaced with Connection Middleware, which is similar to HTTP Middleware in the ASP.NET Core pipeline but for lower-level connections.
+* The Kestrel transport layer has been exposed as a public interface in `Connections.Abstractions`.
+* Ambiguity between headers and trailers has been resolved by moving trailing headers to a new collection.
+* Synchronous IO APIs, such as `HttpReqeuest.Body.Read`, are a common source of thread starvation leading to app crashes. In 3.0, `AllowSynchronousIO` is disabled by default.
+
+For more information, see <xref:migration/22-to-30>.
+
+## Hosting EventSourceProvider new request counters
+
+The Hosting EventSourceProvider (Microsoft.AspNetCore.Hosting) emits the following request counters:
+
+* `requests-per-second`
+* `total-requests`
+* `current-requests`
+* `failed-requests`
+
+## Endpoint routing
+
+See https://github.com/aspnet/AspNetCore.Docs/issues/14291
+
+## Health Checks
+
+Health Checks use endpoint routing with the Generic Host. In `Startup.Configure`, call `MapHealthChecks` on the endpoint builder with the endpoint URL or relative path:
+
+```csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health");
+});
+```
+
+Health Checks endpoints can:
+
+* Specify one or more permitted hosts/ports.
+* Require authorization.
+* Require CORS.
+
+For more information, see the following articles:
+
+* <xref:migration/22-to-30#health-checks>
+* <xref:host-and-deploy/health-checks>
+
+## Pipes on HttpContext
+
+TODO by PU
+
+<!-- indirectly related, https://github.com/dotnet/docs/pull/14414 won't be published by 9/23  -->
+
+## Worker service and SDK
+
+See https://github.com/aspnet/AspNetCore.Docs/issues/14269
+
+## Forwarded Headers Middleware improvements
+
+In previous versions, calling <xref:Microsoft.AspNetCore.Builder.HstsBuilderExtensions.UseHsts*> and  <xref:Microsoft.AspNetCore.Builder.HttpsPolicyBuilderExtensions.UseHttpsRedirection*> were problematic when deployed to an Azure Linux or behind any reverse proxy other than IIS. The fix for previous versions is documented in [Forward the scheme for Linux and non-IIS reverse proxies](xref:host-and-deploy/proxy-load-balancer#forward-the-scheme-for-linux-and-non-iis-reverse-proxies).
+
+In ASP.NET Core 3.0, this problem has been fixed. The host enables the [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer#forwarded-headers-middleware-options) when the `ASPNETCORE_FORWARDEDHEADERS_ENABLED` environment variable has been set to `true`.
+
+## Additional information
```

Comment:
I like this. This is really nitpicky, but "HTTP/2 is now enabled by default in Kestrel for HTTPS endpoints." is a little more accurate since the protocol is configured on a per-endpoint basis.

Reply:
I like that modification. Will update my "suggestion"